packages/react-sdk/src/useObservable.ts

```diff
@@ -2,31 +2,45 @@
 import { useCallback, useEffect, useMemo, useState } from 'react'
 import { type Observable, type PersistedModel } from '@river-build/sdk'
 
-export type ObservableConfig<T> = {
-    fireImmediately?: boolean
-    onUpdate?: (data: T) => void
-    onError?: (error: Error) => void
-    onSaved?: (data: T) => void
-}
-
-type ObservableReturn<T> = {
-    data: T | undefined
-    error: Error | undefined
-    status: PersistedModel<T>['status']
-    isLoading: boolean
-    isError: boolean
-    isSaving: boolean
-    isSaved: boolean
-    isLoaded: boolean
-}
+export type ObservableConfig<Data> = Data extends PersistedModel<infer UnwrappedData>
+    ? {
+          fireImmediately?: boolean
+          onUpdate?: (data: UnwrappedData) => void
+          onError?: (error: Error) => void
+          onSaved?: (data: UnwrappedData) => void
+      }
+    : {
+          fireImmediately?: boolean
+          onUpdate?: (data: Data) => void
+          onError?: (error: Error) => void
+          onSaved?: (data: Data) => void
```

Comment:
in the future, I want to remove those functions if the Observable is not a persisted one.
You won't hit `onSaved` if you're not saving anything, the types should tell this.